recipe/build.sh

```diff
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+export PATH="$PATH:$BUILD_PREFIX/nvvm/bin/"
+
```

Comment:
Am curious if we still need this after the CMake changes

```suggestion
# todo: see if we can drop this and still find `cicc`
# export PATH="$PATH:$BUILD_PREFIX/nvvm/bin/"

```

Reply:
The above changes won't fix this, as CMake isn't what is invoking `cicc`. This points to some issue with the env

I thought `nvcc` needs to be in the HOST not BUILD as it doesn't change when cross-compiling

Reply:
`nvcc` needs to be in `requirements/build` as it is a build tool

Independent of that `nvcc` still has some content that that could depend on the target architecture ( https://github.com/conda-forge/cuda-nvcc-impl-feedstock/issues/4 )

We do [add `nvvm/bin` to the path in `nvcc.profile`]( https://github.com/conda-forge/cuda-nvcc-impl-feedstock/blob/4199d2047d7ada2c7c2c1fdb84f0fe57f860cd34/recipe/nvcc.profile.patch#L12 ). Is it possible this path needs to be changed? If so, is there a good way to check what `nvcc` thinks this path is now?

Reply:
> nvcc needs to be in requirements/build as it is a build tool

:+1:  I must be mixing things up. Looking at other projects the bigger difference is they are using `{{ compiler('cuda') }}` and not directly the `cuda-nvcc` package. That might the source of the issue

Reply:
All good. It can be confusing 😅

Yeah `{{ compiler('cuda') }}` is Jinja `{{  }}` with a special function `compiler`, which takes the `str` provided (in this case `cuda`) and appends `_compiler` to that to form the variant key (so `cuda_compiler`). It then uses this variant key to look for what `cuda_compiler` should be. There are [a few places it will look for this variant key]( https://docs.conda.io/projects/conda-build/en/latest/resources/variants.html#creating-conda-build-variant-config-files ) one being `conda_build_config.yaml`, which in this case specifies these options:

https://github.com/scopetools/cudadecon/blob/937942579292bab46575039795721f8b75e0a4e8/recipe/conda_build_config.yaml#L16-L18

`nvcc` refers to [conda-forge/nvcc-feedstock]( https://github.com/conda-forge/nvcc-feedstock ), which we used for CUDA 11 builds. `cuda-nvcc` refers to [conda-forge/cuda-nvcc-feedstock]( https://github.com/conda-forge/cuda-nvcc-feedstock ), which we have added for CUDA 12. Depending on whether a CUDA 11 or 12 build is done, it will pick from these

There is one last step where conda-build converts the `cuda_compiler` selected into the compiler used, which is it adds `_{{ target_platform }}` to it. IOW `nvcc` would become `nvcc_linux-64` for builds targeting Linux x86_64 on CUDA 11. Or `cuda-nvcc` would become `cuda-nvcc_linux-aarch64` for builds targeting Linux ARM 64-bit. The package installed will be selected to run on the build platform

Hopefully that makes more sense